pkg/controller/replication/replication_controller.go

```diff
@@ -478,10 +573,19 @@ func (rm *ReplicationManager) manageReplicas(filteredPods []*api.Pod, rc *api.Re
 		var wg sync.WaitGroup
 		wg.Add(diff)
 		glog.V(2).Infof("Too few %q/%q replicas, need %d, creating %d", rc.Namespace, rc.Name, rc.Spec.Replicas, diff)
+		// needs to set the gvk to generate ownerReference
+		oldKind := rc.Kind
+		oldAPIVersion := rc.APIVersion
+		defer func() {
+			rc.Kind = oldKind
+			rc.APIVersion = oldAPIVersion
+		}()
+		rc.APIVersion = v1.SchemeGroupVersion.String()
+		rc.Kind = "ReplicationController"
```

Comment:
Per offline discussion, will just pass in a hard-coded GroupVersionKind instead of setting in the object.

Considered alternatives: 1. Parsing from self link will get us the resource name, not kind name. 2. consulting the schema may get unstable result because a type can be registered in multiple groups/versions.